dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/exception/ExprTypeMismatchException.java

```diff
@@ -0,0 +1,8 @@
+package com.dat3m.dartagnan.parsers.program.exception;
+
+public class ExprTypeMismatchException extends RuntimeException {
```

Comment:
I think this error type goes beyond encoding. I consider `ExprTypeMismatchException` as a `MalformedProgramExcpetion `

Reply:
I agree that it may be seen as a `MalformedProgramException` if the types of the program are not cast properly. But I'm wondering when this type check happens? If this exception is only thrown during program construction, then it can be a `MalformedProgramException`, if it is only ever thrown during encoding, it should be a `EncodingException`, if it can be thrown in both cases, it can stay as it is (i.e. inherit from `RuntimeException`).

Reply:
This is now checked using `Preconditions.check()` in the constructor of `IExprBin`